narayana-spring-boot-core/pom.xml

```diff
@@ -60,6 +60,11 @@
         <artifactId>spring-boot-configuration-processor</artifactId>
         <optional>true</optional>
     </dependency>
+    <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-jdbc</artifactId>
+        <optional>true</optional>
+    </dependency>
```

Comment:
How come this is needed?

Reply:
I'm using Spring API to determine database type (`JdbcUtils.commonDatabaseName(...)`) in `GenericXADataSourceWrapper.java`

Reply:
But then why is the dependency optional?

Reply:
You may use narayana-spring-boot for JMS and JTA only so no need for JDBC dependencies. This is a core library, it's a kind of perspective.

Reply:
Sure, but how is codepath that potentially uses `JdbcUtils` guarded against the case of using being used without `spring-jdbc`?

Reply:
Well, SpringBoot won't initialize `DataSource` without spring-jdbc dependency. It's part of there AutoConfiguration.